addons/binding/org.openhab.binding.foobot/src/main/java/org/openhab/binding/foobot/handler/FoobotHandler.java

```diff
@@ -0,0 +1,284 @@
+        try {
+            value = getValue(channelId, foobotResponse);
+        } catch (Exception e) {
+            logger.debug("Foobot device doesn't provide sensor data for channel id = {}", channelId.toUpperCase());
+        }
+
+        if (value == null) {
+            updateState(channelId, UnDefType.NULL);
+            logger.debug("Update channel {}: null value for the channel", channelId);
+        } else if (value instanceof BigDecimal) {
+            State state = new DecimalType((BigDecimal) value);
+            updateState(channelId, state);
+            logger.debug("Update channel {} with state {} ({})", channelId, state, value.getClass().getSimpleName());
+        } else {
+            updateState(channelId, UnDefType.UNDEF);
+            logger.debug("Update channel {}: Unsupported value type {}", channelId, value.getClass().getSimpleName());
+        }
+    }
+
+    private FoobotJsonData updateFoobotData() {
+        FoobotJsonData result = null;
+        try {
+            // Run the HTTP request and get the list of all foobot devices belonging to the user from foobot.io
+            String urlStr = URL2.replace("%uuid%", StringUtils.trimToEmpty(uuid));
+            urlStr = urlStr.replace("%sensors%", StringUtils.trimToEmpty(""));
+            logger.debug("URL = {}", urlStr);
+
+            // Run the HTTP request and get the JSON response from foobot.io
+            HttpClient httpClient = new DefaultHttpClient();
+            HttpGet getRequest = new HttpGet(urlStr);
+            getRequest.addHeader("accept", "application/json");
+            getRequest.addHeader("X-API-KEY-TOKEN", config.apikey);
+            HttpResponse response = httpClient.execute(getRequest);
+
+            String responseData = EntityUtils.toString(response.getEntity());
+            logger.debug(responseData);
+
+            if (StringUtils.trimToNull(responseData) == null) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR, "no data returned");
+            } else {
+                // Map the JSON response to an object
+                result = gson.fromJson(responseData, FoobotJsonData.class);
+                updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
+            }
+        } catch (MalformedURLException e) {
+            logger.debug("Constructed url is not valid: {}", e.getMessage());
+        } catch (JsonSyntaxException e) {
+            logger.debug("Error running foobot.io request: {}", e.getMessage());
+        } catch (IOException | IllegalStateException e) {
+            logger.debug(e.getMessage());
+        }
+
+        return result;
+    }
+
+    public static BigDecimal getValue(String channelId, FoobotJsonData data) {
+        if (data == null) {
+            return null;
+        }
+        switch (channelId) {
+            case FoobotBindingConstants.TMP:
+                return new BigDecimal(data.getDatapointsList().get(data.getSensors().indexOf("tmp")));
```

Comment:
I think you could easily split of a method which can return that BigDecimal given a data and label, you could even consider storing a mapping between the channelId and the label in the sensors list, then you can base the rest of the code on the map which is pretty straight forward.